### PR TITLE
Toggle subscription

### PR DESCRIPTION
**Pull request by bkuhl**
* Added ability to manage the subscription for a given game

The gif is a bit more dated, now it actually hits the ajax endpoints.  There's not a good way for me to demo that it's actually subscribed other than referencing the tests.

vuejs-demo

### CODE REVIEW

File: app/Guild.php [hunk 1]
```diff
@@ -73,4 +73,8 @@ public function subscriptions() : HasMany
     {
         return $this->hasMany(Subscription::class);
     }
+
+    public function isSubscribed(Game $game, int $interest)
+    {
```
Comment (alexknutson) [inline app/Guild.php]: is this needed?

Comment (bkuhl) [reply app/Guild.php]: Doh, removed.

File: app/Providers/AppServiceProvider.php [hunk 2]
```diff
@@ -27,6 +29,17 @@ public function boot()
...
+
+        /*
+         * Directives are used for ensuring javascript data gets passed in a <script>
+         * // outside of the main page's vuejs template
+         */
+        Blade::directive('jsData', function () {
+            return '<?php ob_start(); ?>';
+        });
+        Blade::directive('endjsData', function () {
+            return '<?php \\'.Html::class.'::$jsData .= ob_get_clean(); ?>';
```
Comment (alexknutson) [inline app/Providers/AppServiceProvider.php]: `ob_start` and `ob_get_clean`?

Comment (bkuhl) [reply app/Providers/AppServiceProvider.php]: ob = output buffer which catches the output and puts it into memory.  `ob_get_clean()` takes what it is in memory and returns it, allowing you to assign it to a variable.

Essentially the need here is that when we pass data to javascript, it needs to be at the end of the page, just before <script> or VueJS complains.  These directives allow us to do that by adding something like this to our template, anywhere we want without having to worry about order:

```
@jsData
 var myVar = {!! json_encode($myValue) !!}
@endjsData
```

Comment (alexknutson) [reply app/Providers/AppServiceProvider.php]: :+1:

**Pull request merged**